src/main/java/com/it4045/pokemonviewer/enterprise/dto/Pokemon.java

```diff
@@ -2,35 +2,48 @@
 
 import lombok.Data;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public @Data
 class Pokemon {
 
-    private int pokemonId;
-    private String pokemonNumber;
-    private String pokemonName;
-    private String pokemonImg;
-    private List<String> pokemonType;
-    private String pokemonHeight;
-    private String pokemonWeight;
-    private String pokemonCandy;
-    private String pokemonEgg;
-    private double pokemonSpawnChoice;
-    private double pokemonAvgSpawns;
-    private String pokemonSpawnTime;
-    private List<Double> pokemonMultipliers;
-    private List<String> pokemonWeakness;
-    private List<Evolution> pokemonNextEvolution;
-    private List<Evolution> pokemonPrevEvolution;
+    private int id;
+    private String number;
+    private String name;
+    private String img;
+    private List<String> type;
+    private String height;
+    private String weight;
+    private String candy;
+    private String egg;
+    private double spawnChoice;
+    private double avgSpawns;
+    private String spawnTime;
+    private List<Double> multipliers;
+    private List<String> weakness;
+    private List<Map.Entry<Pokemon, String>> previousEvolution;
+    private List<Map.Entry<Pokemon, String>> nextEvolution;
```

Comment:
First technical suggestion: variable names should reflect access modifiers.

Here, I changed the names of the variables in the Pokemon class to be simplified, without using the "pokemon" prefix. The prefix would be a good idea if these were public variables, but because they are private, they will never be accessed outside of a Pokemon object. Changing them in this way should make the Pokemon class a bit more concise and easy to read at a glance.

Reply:
Agreed.  No need to replicate the class name inside of each attribute name.  That's redundant, as one can tell the context from the class name.